src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -90,18 +90,42 @@ public async Task StartContainersAsync(IExecutionContext executionContext, objec
             ArgUtil.NotNull(dockerVersion.ClientVersion, nameof(dockerVersion.ClientVersion));
 
 #if OS_WINDOWS
-            Version requiredDockerVersion = new Version(17, 6);
+            Version requiredDockerCeVersion = new Version(17, 6);
 #else
-            Version requiredDockerVersion = new Version(17, 12);
+            Version requiredDockerCeVersion = new Version(17, 12);
 #endif
+            Version requiredDockerMsMobyVersion = new Version(3, 0);
```

Comment:
Instead of having 2 variables, perhaps you could override the `requiredDockerVersion` instead based on the `isMobyInstalled` value. Since `isMobyInstalled`  is used only for the toggle between these two variables, overriding the earlier value would simplify the logic, do you agree?

Reply:
Here we want to validate moby version if current docker version less than dockerce version.